src/Analyzer.JsonRuleEngine/JsonRuleEvaluation.cs

```diff
@@ -74,6 +74,26 @@ public bool HasResults
             get => Results.Any() || Evaluations.Any(e => e.HasResults);
         }
 
+        /// <summary>
+        /// Whether or not there are any evaluations associated with this <see cref="JsonRuleEvaluation"/>.
+        /// </summary>
+        /// <returns>True if there are any evaluations in this <see cref="JsonRuleEvaluation"/>.
+        /// False otherwise.</returns>
+        public bool HasEvaluations
+        {
+            get => Evaluations.Any();
+        }
+
+        /// <summary>
+        /// Whether or not there are any results or evaluations associated with this <see cref="JsonRuleEvaluation"/>.
+        /// </summary>
+        /// <returns>True if there are no results and no evaluations in this <see cref="JsonRuleEvaluation"/> or a sub-<see cref="JsonRuleEvaluation"/>.
+        /// False otherwise.</returns>
+        public bool NoScopesFound
+        {
+            get => !HasResults && !HasEvaluations;
+        }
```

Comment:
Just checking `HasResults` is sufficient.  Actually, checking `HasEvaluations` as well I think is a bug, since there might be nested evaluations without results.

I think `NoScopesFound` isn't needed, since `HasResults` already provides everything needed.